tidy3d/plugins/mode/mode_solver.py

```diff
@@ -339,6 +353,339 @@ def data_raw(self) -> ModeSolverData:
+
+        rotated_structures = self._rotate_structure()
+        rotated_simulation = self.simulation.updated_copy(structures=rotated_structures)
+        rotated_mode_spec = self.mode_spec.updated_copy(
+            bend_angle_rotation=False, angle_theta=0, angle_phi=0
+        )
+
+        return self.updated_copy(simulation=rotated_simulation, mode_spec=rotated_mode_spec)
+
+    def _rotate_structure(self) -> List[Structure]:
+        "rotate the structures intersecting with modal plane by angle theta if bend_correction"
+        "is enabeled for bend simulations."
+
+        _, (idx_u, idx_v) = self.plane.pop_axis((0, 1, 2), axis=self.bend_axis_3d)
+
+        mnt_center = self.plane.center
+        angle_theta = self.mode_spec.angle_theta
+        angle_phi = self.mode_spec.angle_phi
+
+        theta_map = {
+            (0, 2): -angle_theta * np.cos(angle_phi),
+            (0, 1): angle_theta * np.sin(angle_phi),
+            (1, 2): angle_theta * np.cos(angle_phi),
+            (1, 0): -angle_theta * np.sin(angle_phi),
+            (2, 1): -angle_theta * np.cos(angle_phi),
+            (2, 0): angle_theta * np.sin(angle_phi),
+        }
+        theta = theta_map.get((self.normal_axis, self.bend_axis_3d), 0)
+
+        # Get the translation values
+        translate_coords = [0, 0, 0]
+        translate_coords[idx_u] = mnt_center[idx_u]
+        translate_coords[idx_v] = mnt_center[idx_v]
+
+        rotated_structures = []
+        for structure in Scene.intersecting_structures(self.plane, self.simulation.structures):
+            # Rotate and apply translations
+            geometry = structure.geometry
+            geometry = (
+                geometry.translated(
+                    x=-translate_coords[0], y=-translate_coords[1], z=-translate_coords[2]
+                )
+                .rotated(theta, axis=self.bend_axis_3d)
+                .translated(x=translate_coords[0], y=translate_coords[1], z=translate_coords[2])
+            )
+
+            rotated_structures.append(structure.updated_copy(geometry=geometry))
+
+        return rotated_structures
+
+    @cached_property
+    def rotated_bend_center(self) -> List:
+        """Calculate the center of the rotated bend such that the modal plane is normal
+        to the azimuthal direction of the bend."""
+        rotated_bend_center = list(self.plane.center)
+        idx_rotate = 3 - self.normal_axis - self.bend_axis_3d
+        rotated_bend_center[idx_rotate] -= self.mode_spec.bend_radius
+
+        return rotated_bend_center
+
+    def _car_2_cyn(
+        self, mode_solver_data: ModeSolverData
+    ) -> Dict[Union[ScalarModeFieldCylindricalDataArray, ModeIndexDataArray]]:
+        """Convert Cartesian field data to cylindrical coordinates centered the
+        rotated bend center."""
+
+        # Extract coordinate values from one of the six field components
+        pts = [mode_solver_data.Ex[name].values.copy() for name in ["x", "y", "z"]]
```

Comment:
I can't exactly follow what's happening in this function: do results depend on the specific choice to pick Ex?